src/org/javarosa/core/model/FormDef.java

```diff
@@ -982,6 +982,9 @@ public String fillTemplateString(String template, TreeReference contextRef,
      * @param itemset The binding for an itemset, where the choices will be populated
      * @param curQRef A reference to the current question's element, which will be
      *                used to determine the values to be chosen from.
+     *
+     * Side effects: Mutates the itemset object passed by clearing any choices currently on it, and populating with
+     * choices based on the question reference curQRef that was passed.
```

Comment:
Hi!

1. Since the return type is void, by definition this method does its work via side effects. So, no need to mention side effects.
2. I don’t think a note like this should _follow_ the @param section.
3. Above, we already have “Will modify the itemset ...”. Maybe you just want to improve that.

Reply:
@dcbriccetti Thanks for the response!

Regarding (1): 

**dcbriccetti wrote:**
>1. Since the return type is void, by definition this method does its work via side effects. So, no need to mention side effects.

Regarding this, what of Josh Bloch's recommendation regarding documentation [at this point in this video](https://youtu.be/heh4OeB9A-c?t=2036) (not sure where in the vid the specific example is)?

**Josh Bloch wrote:**
>  ...and any side effects. Those are particularly important. If you don't document side effects, people will shoot themselves in the foot. I'll give you an example of this later

2. I agree.

3. That makes sense to me as well; I'll do that. I didn't notice it.

Regarding "`Side effects: `", I tried putting that in the documentation at work, and people didn't like it either. For me, I try to write as many pure functions as I can, and I like it because it's explicit (even though "`void`" is also explicit). It's probably better to lean towards this in looser languages, like Python, which don't declare a void function.

Reply:
I believe I have taken care of the issues pointed out. Marking resolved!